src/schema/v2/userAccessibleProperties.ts

```diff
@@ -0,0 +1,46 @@
+import { GraphQLEnumType, GraphQLUnionType } from "graphql"
+import { connectionWithCursorInfo } from "./fields/pagination"
+import { ArtworkType } from "./artwork"
+import { ProfileType } from "./profile"
+import { PartnerType } from "./partner"
+import { ArtistType } from "./artist"
+
+const userAccessControlPropertiesType = new GraphQLUnionType({
+  name: "UserAccessibleProperty",
+  types: () => [ArtistType, ArtworkType, ProfileType, PartnerType],
+  resolveType: (object) => {
+    if ("birthday" in object) {
```

Comment:
This sequence of `if` statements is kind of silly. The Gravity API should return `property_type` in the JSON for an access control (https://github.com/artsy/gravity/blob/ca7a46948f4b1acad7282679d7dc1446a5499c71/app/models/domain/access_control.rb#L21-L24).

That way, we know what the 'type' of the property is in a more sensible way than this. This currently introspects the JSON returned as `property` from that API, and tries to deduce what the type is based on the fields available (kind of brittle and feels not great). Would be much better if `property_type: "Artist"` was included, for instance (can do that Monday).